File: apps/web/refactor.py
import os
import re

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    new_content = re.sub(r'catch\s*\(\s*err\s*:\s*any\s*\)', 'catch (err: unknown)', content)
    new_content = re.sub(r'catch\s*\(\s*e\s*:\s*any\s*\)', 'catch (e: unknown)', new_content)
    new_content = re.sub(r'catch\s*\(\s*error\s*:\s*any\s*\)', 'catch (error: unknown)', new_content)
    
    # Replacing common error handling pattern (err as any).message etc.
    new_content = new_content.replace('err.response?.data?.message', '(err as any).response?.data?.message')
    
    if new_content != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print('Updated', filepath)

src_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'src')
for root, _, files in os.walk(src_dir):
    for file in files:
        if file.endswith('.ts') or file.endswith('.tsx'):
            process_file(os.path.join(root, file))

File: apps/web/refactor.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
  let results = [];
  const list = fs.readdirSync(dir);
  list.forEach(function(file) {
    file = dir + '/' + file;
    const stat = fs.statSync(file);
    if (stat && stat.isDirectory()) { 
      results = results.concat(walk(file));
    } else { 
      if (file.endsWith('.ts') || file.endsWith('.tsx')) {
        results.push(file);
      }
    }
  });
  return results;
}

const srcDir = path.join(__dirname, 'src');
const files = walk(srcDir);

files.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  let newContent = content
    .replace(/catch\s*\(\s*err\s*:\s*any\s*\)/g, 'catch (err: unknown)')
    .replace(/catch\s*\(\s*e\s*:\s*any\s*\)/g, 'catch (e: unknown)')
    .replace(/catch\s*\(\s*error\s*:\s*any\s*\)/g, 'catch (error: unknown)')
    .replace(/err\.response\?\.data\?\.message/g, '(err as any).response?.data?.message')
    .replace(/err\ instanceof\ Error\ \?\ err\.message\ :\ '([^']+)'/g, 'err instanceof Error ? err.message : \'$1\'');
  
  if (content !== newContent) {
    fs.writeFileSync(file, newContent, 'utf8');
    console.log('Updated', file);
  }
});
